### PR TITLE
Fix small p2p network issues

### PR DESCRIPTION
**Pull request by chimp1984**

### CODE REVIEW

**Comment by ripcurlx** [comment]
@chimp1984 I've added [WIP] to your title, so it is easier to see that you are still working on this.

**Review by ripcurlx — APPROVED** [review]
utACK - Looks good to me. Thanks for the PR! 👍

**Comment by ripcurlx** [comment]
Maybe @sqrrm or @ManfredKarrer could give it a quick ACK again. From my side changes seem to be fine.

**Comment by christophsturm** [comment]
looks good. some of the commits messages could be improved by explaining why the change was done, and also the pr description could explain the overall motivation.

**Comment by ripcurlx** [comment]
> looks good. some of the commits messages could be improved by explaining why the change was done, and also the pr description could explain the overall motivation.

@chimp1984 Yes, please just add a small description for your PRs in the future. Everything looks perfectly fine, what you are contributing, but it helps to review if there is more context.

**Review by freimair — APPROVED** [review]
ACK.

though, next time, please leave the formatting as is. it is very hard to spot the differences that actually change something.

**Comment by christophsturm** [comment]
> though, next time, please leave the formatting as is. it is very hard to spot the differences that actually change something.

also a good idea: reformat in a separate commit

**Comment by chimp1984** [comment]
> the commits messages could be improved by explaining why the change was done, and also the pr description could explain the overall motivation.

We do better next time.

> please leave the formatting as is.

I use the formatting settings from the IDEA which are part of the project and as far I understand formatting should be applied before commit.

> reformat in a separate commit

Agree.

File: p2p/src/main/java/bisq/network/p2p/storage/messages/AddOncePayload.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.network.p2p.storage.messages;
+
+/**
+ * Marker interface for messages which must not be added again after a remove message has been received (e.g. MailboxMessages).
+ */
+public interface AddOncePayload {
```
Comment (julianknutsen) [inline p2p/src/main/java/bisq/network/p2p/storage/messages/AddOncePayload.java:23]: Is there an issue number that this change fixes? After reviewing and refactoring this code I'm having trouble understanding the bug that would have needed this fix. I'd like to try and reproduce it in my tests, if possible.

**Pull request merged**